### PR TITLE
tests: Fix cookies

### PR DESCRIPTION
**Pull request by rhansen**
Multiple commits:
* tests: Refine `helper/multipleUsers.js`
* tests: Promisify `authorship_of_editions.js`
* tests: Promisify `multiple_authors_clear_authorship_colors.js`
* tests: Promisify some of the `helper.js` tests
* tests: Fix encoding of prefs cookie
* tests: Always set cookie path to / (to match non-test behavior)
* tests: Fix cookie name in `helper.js` tests
* pad_cookie: Move initial cookie read+save to `init()`
* pad_cookie: Re-read prefs cookie on every call to `getPref()`
* tests: Use cookie libraries to manipulate cookies
* tests: Wait for commit rather than sleep

### CODE REVIEW

File: src/tests/frontend/specs/authorship_of_editions.js [hunk 1]
```diff
@@ -6,79 +6,76 @@ describe('author of pad edition', function () {
...
+    $insertOrderedListButton.click();
+
+    await helper.waitForPromise(() => {
+      const $lineWithOrderedList = getLine(LINE_WITH_ORDERED_LIST);
+      return $lineWithOrderedList.find('ol li').length === 1;
+    });
+
+    // Need a timeout here to make sure all changes were saved.
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
```
Comment (webzwo0i) [inline src/tests/frontend/specs/authorship_of_editions.js:51]: I would recommend using helper.commits here.
```
diff --git a/src/tests/frontend/specs/authorship_of_editions.js b/src/tests/frontend/specs/authorship_of_editions.js
index f78b9d1e..b0724419 100644
--- a/src/tests/frontend/specs/authorship_of_editions.js
+++ b/src/tests/frontend/specs/authorship_of_editions.js
@@ -19,7 +19,8 @@ describe('author of pad edition', function () {
     // wait for lines to be processed by Etherpad
     await helper.waitForPromise(() => {
       const $lineWithUnorderedList = getLine(LINE_WITH_UNORDERED_LIST);
-      return $lineWithUnorderedList.text() === 'line with unordered list';
+      return $lineWithUnorderedList.text() === 'line with unordered list' &&
+        helper.commits.length === 1;
     });
 
     // create the unordered list
@@ -31,7 +32,8 @@ describe('author of pad edition', function () {
 
     await helper.waitForPromise(() => {
       const $lineWithUnorderedList = getLine(LINE_WITH_UNORDERED_LIST);
-      return $lineWithUnorderedList.find('ul li').length === 1;
+      return $lineWithUnorderedList.find('ul li').length === 1 &&
+        helper.commits.length === 2;
     });
 
     // create the ordered list
@@ -43,12 +45,10 @@ describe('author of pad edition', function () {
 
     await helper.waitForPromise(() => {
       const $lineWithOrderedList = getLine(LINE_WITH_ORDERED_LIST);
-      return $lineWithOrderedList.find('ol li').length === 1;
+      return $lineWithOrderedList.find('ol li').length === 1 &&
+        helper.commits.length === 3;
     });
 
-    // Need a timeout here to make sure all changes were saved.
-    await new Promise((resolve) => setTimeout(resolve, 1000));
-
     // Expire cookie, so author is changed after reloading the pad.
     const {Cookies} = helper.padChrome$.window.require('ep_etherpad-lite/static/js/pad_utils');
     Cookies.remove('token');
```

File: src/tests/frontend/specs/multiple_authors_clear_authorship_colors.js [hunk 2]
```diff
@@ -2,36 +2,32 @@
...
+
+    // make sure pad has at least 3 lines
+    const $firstLine = helper.padInner$('div').first();
+    $firstLine.html('Hello World');
+
+    // wait for lines to be processed by Etherpad
+    await helper.waitForPromise(() => $firstLine.text() === 'Hello World');
+
+    // Need a timeout here to make sure all changes were saved.
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (webzwo0i) [inline src/tests/frontend/specs/multiple_authors_clear_authorship_colors.js:17]: same here, better use helper.commits, as it is more explicit

File: src/tests/frontend/specs/authorship_of_editions.js (hunk 1 above)
Comment (rhansen) [reply src/tests/frontend/specs/authorship_of_editions.js:45]: Good idea, thanks!

Done.

File: src/tests/frontend/specs/multiple_authors_clear_authorship_colors.js (hunk 2 above)
Comment (rhansen) [reply src/tests/frontend/specs/multiple_authors_clear_authorship_colors.js]: Done.

**Pull request merged**